### PR TITLE
Add txHash to events

### PR DESCRIPTION
**Pull request by michalsidzej**
Resolves L2B-3032

### CODE REVIEW

File: packages/backend/src/peripherals/database/EventRepository.ts [hunk 1]
```diff
@@ -27,19 +28,20 @@ export class EventRepository extends BaseRepository {
...
-    const rows = await knex('events')
+    const blockNumbers = await knex('events')
       .where('chain_id', Number(chainId))
       .andWhere('block_number', '>', fromBlock)
       .andWhere('block_number', '<=', toBlock)
+      .distinct('block_number')
       .orderBy('block_number', 'asc')
 
-    return rows.map(toRecord)
+    return blockNumbers.map((row) => row.block_number)
```
Comment (sdlyy) [inline packages/backend/src/peripherals/database/EventRepository.ts:44]: Couldn't we leave old func?

File: packages/backend/src/indexers/EventIndexer.ts [hunk 2]
```diff
@@ -128,8 +160,7 @@ export class EventIndexer extends ChildIndexer {
   }
 
   protected override async invalidate(targetHeight: number): Promise<number> {
-    const blockNumber = Math.max(targetHeight, this.startBlock)
-    await this.eventRepository.deleteAfter(blockNumber, this.chainId)
+    await this.eventRepository.deleteAfter(targetHeight, this.chainId)
```
Comment (sdlyy) [inline packages/backend/src/indexers/EventIndexer.ts:163]: What's up with removing start block ❔

**Review by sdlyy — APPROVED** [review]

File: packages/backend/src/indexers/EventIndexer.ts (hunk 2 above)
Comment (michalsidzej) [reply packages/backend/src/indexers/EventIndexer.ts:163]: The problem was that we would not remove the startBlock record when trying to invalidate to 0. This was a bug. Also, we can actually invalidate to 0, as the update will start from `startBlock` anyway, and this way we're sure that everything is clean.

File: packages/backend/src/peripherals/database/EventRepository.ts (hunk 1 above)
Comment (michalsidzej) [reply packages/backend/src/peripherals/database/EventRepository.ts:44]: The thing is that there will be block numbers with multiple records (when there are multiple txs in a block). This func ensures that we get distinct block numbers and it's also easy to test.

**Pull request merged**